crates/turbopack-ecmascript/src/references/esm/url.rs

```diff
@@ -149,177 +113,71 @@ impl CodeGenerateable for UrlAssetReference {
         let this = self_vc.await?;
         let mut visitors = vec![];
 
-        let inner_asset = self_vc.inner_asset().await?;
-
-        if let Some(inner) = &*inner_asset {
-            let chunk_item = inner.as_chunk_item(context);
-
-            // We rewrite the first `new URL()` arguments to be a require() of the chunk
-            // item, which exports the static asset path to the linked file.
-            let id = chunk_item.id().await?;
-
-            // In Browser environments, we rewrite the `import.meta.url` to be a
-            // location.origin because it allows us to access files from the
-            // root of the dev server. In node env, the `import.meta.url` already be the correct `file://` URL to load files.
-            let is_browser = *this.is_browser.await?;
+        let referenced_asset = self_vc.get_referenced_asset().await?;
+
+        match &*referenced_asset {
+            ReferencedAsset::Some(asset) => {
+                // We rewrite the first `new URL()` arguments to be a require() of the chunk
+                // item, which exports the static asset path to the linked file.
+                let id = asset.as_chunk_item(context).id().await?;
+                // In Browser environments, we rewrite the `import.meta.url` to be a
+                // location.origin because it allows us to access files from the
+                // root of the dev server. In node env, the `import.meta.url` already be the correct `file://` URL to load files.
+                let is_browser = *this.is_browser.await?;
+
+                let ast_path = this.ast_path.await?;
+                visitors.push(
+                    create_visitor!(ast_path, visit_mut_expr(new_expr: &mut Expr) {
+                        if let Expr::New(NewExpr { args: Some(args), .. }) = new_expr {
+                            if let Some(ExprOrSpread { box expr, spread: None }) = args.get_mut(0) {
+                                *expr = quote!(
+                                    "__turbopack_require__($id)" as Expr,
+                                    id: Expr = module_id_to_lit(&id),
+                                );
+                            }
 
-            let ast_path = this.ast_path.await?;
-            visitors.push(
-                create_visitor!(ast_path, visit_mut_expr(new_expr: &mut Expr) {
-                    if let Expr::New(NewExpr { args: Some(args), .. }) = new_expr {
-                        if let Some(ExprOrSpread { box expr, spread: None }) = args.get_mut(0) {
-                            *expr = quote!(
-                                "__turbopack_require__($id)" as Expr,
-                                id: Expr = module_id_to_lit(&id),
-                            );
+                            if is_browser {
+                                if let Some(ExprOrSpread { box expr, spread: None }) = args.get_mut(1) {
+                                    *expr = quote!("location.href" as Expr);
```

Comment:
I typo'd this (but got it right in my comment explanations):
```suggestion
                                    *expr = quote!("location.origin" as Expr);
```